examples/simulation/plot_stc_metrics.py

```diff
@@ -0,0 +1,264 @@
+
+# Region
+events = mne.find_events(raw_region, initial_event=True)
+tmax = (len(source_time_series) - 1) * tstep
+epochs = mne.Epochs(raw_region, events, 1, tmin=0, tmax=tmax)
+evoked_region = epochs.average()
+
+# Dipole
+events = mne.find_events(raw_dipole, initial_event=True)
+tmax = (len(source_time_series) - 1) * tstep
+epochs = mne.Epochs(raw_dipole, events, 1, tmin=0, tmax=tmax)
+evoked_dipole = epochs.average()
+
+###############################################################################
+# Create true stcs corresponding to evoked
+# ----------------------------------------
+#
+# Before we computed stcs corresponding to raw data. To be able to compare
+# it with the reconstruction, based on the evoked, true stc should have the
+# same number of time samples.
+
+# Region
+stc_true_region = \
+    source_simulator_region.get_stc(start_sample=0,
+                                    stop_sample=len(source_time_series))
+
+# Dipole
+stc_true_dipole = \
+    source_simulator_dipole.get_stc(start_sample=0,
+                                    stop_sample=len(source_time_series))
+
+###############################################################################
+# Reconstruct simulated sources
+# -----------------------------
+#
+# Compute inverse solution using sLORETA.
+
+# Region
+snr = 30.0
+inv_method = 'sLORETA'
+lambda2 = 1.0 / snr ** 2
+
+inverse_operator = make_inverse_operator(evoked_region.info, fwd, cov,
+                                         loose='auto', depth=0.8,
+                                         fixed=True)
+
+stc_est_region = apply_inverse(evoked_region, inverse_operator, lambda2,
+                               inv_method, pick_ori=None)
+
+# Dipole
+snr = 3.0
+inv_method = 'sLORETA'
+lambda2 = 1.0 / snr ** 2
+
+inverse_operator = make_inverse_operator(evoked_dipole.info, fwd, cov,
+                                         loose='auto', depth=0.8,
+                                         fixed=True)
+
+stc_est_dipole = apply_inverse(evoked_dipole, inverse_operator, lambda2,
+                               inv_method, pick_ori=None)
+
+###############################################################################
+# Compute performance scores for different source amplitude thresholds
+# --------------------------------------------------------------------
+#
+
+thresholds = ['10%', '30%', '50%', '70%', '80%', '90%', '95%', '99%']
+
+###############################################################################
+# For region
+# ^^^^^^^^^^
+#
+
+# create a set of scorers
+scorers = {'RLE': partial(region_localization_error, src=src),
+           'Precision': precision_score, 'Recall': recall_score,
+           'F1 score': f1_score}
+
+# compute results
+region_results = {}
+for name, scorer in scorers.items():
+    region_results[name] = [scorer(stc_true_region, stc_est_region,
+                                   threshold=thx, per_sample=False)
+                            for thx in thresholds]
```

Comment:
if the scorers were taking a list in thresholds `threshold=np.linspace(0.1, 1, N, endpoint=False)` this would simplify 

```py
# create a set of scorers
scorers = {'RLE': partial(region_localization_error, src=src),
           'Precision': precision_score, 'Recall': recall_score,
           'F1 score': f1_score}

# compute results
region_results = {}
thresholds = np.linspace(0.1, 1, N, endpoint=False)
for name, scorer in scorers.items():
    region_results[name] = scorer(stc_true_region, stc_est_region,
                                  thresholds=thresholds, per_sample=False)
```

```py
# create a set of scorers
results = {
  'RLE': region_localization_error( stc_true_region, stc_est_region, src=src,
                                  thresholds=thresholds, per_sample=False),
  'Precision': precision_score(stc_true_region, stc_est_region,
                               thresholds=thresholds, per_sample=False),

...
}
```